### ISSUE TITLE
property_switch_exhaustiveness: False positive on G_PARAM_READABLE prop

### ISSUE THREAD

**Issue by A6GibKm** [bilelmoussaoui/gobject-lint#64]
The file at https://gitlab.gnome.org/GNOME/gtranslator/-/blob/master/src/gtr-po.c reports
```
  gtr_po_set_property (GObject      *object,
                       guint         prop_id,
                       const GValue *value,
                       GParamSpec   *pspec)
  ...
    GtrPo *po = GTR_PO (object);
  ...
  >  switch (prop_id)
  ...
      case PROP_LOCATION:
        gtr_po_set_location (po, G_FILE (g_value_get_object (value)));
  error: Property 'PROP_STATE' should be handled in gtr_po_set_property switch statement
  rule: property_switch_exhaustiveness
```

however, from the fixtures I feel that this is already tested for.

**Comment by bilelmoussaoui** [comment]
This is expected. You should g_assert there and drop the default handler

**Comment by A6GibKm** [comment]
How exactly do you mean?

**Comment by bilelmoussaoui** [comment]
Basically you should use the enum convention rule, re-running it will change your get_property/set_property to have a cast to a typed enum, making the default case useless and must be removed. You should then add any property that shouldn't have a getter or setter to the switch and make it assert so nothing sets it

**Comment by A6GibKm** [comment]
> making the default case useless and must be removed

What prevents one from using a GParamSpec that does not correspond to the GObject, or in general one that does not make sense? I am not sure the default branch is useless.

> You should then add any property that shouldn't have a getter or setter to the switch and make it assert so nothing sets it

Fair, but for the sake of documenting it, is there any project doing this now? I did not find examples in glib, adw, nor gtk after a very brief search.

**Comment by bilelmoussaoui** [comment]
See https://github.com/bilelmoussaoui/goblint/issues/52

**Comment by bilelmoussaoui** [comment]
You can enable legacy style in the config, see https://bilelmoussaoui.github.io/goblint/#property_switch_exhaustiveness and that would remove the warnings you see now.
